Source/Core/Core/FifoPlayer/FifoPlayer.cpp

```diff
@@ -103,6 +103,9 @@ bool FifoPlayer::Play()
 				++m_CurrentFrame;
 			}
 		}
+
+		// Let the waiting thread know we are done leaving
+		PowerPC::FinishStateMove();
```

Comment:
There is no need to call this every iteration of the loop. Just have it after the while loop.